### PR TITLE
Dash reminder

### LINKED ISSUE #17: We should let users know about appending messages with --
We should let users know about appending messages with --, nowhere in the plugin or chat does it let them know about this feature.

### PR DESCRIPTION
**Pull request by sournub**
Fixed #17 ?

I just cannot figure out how to get the reminder to send after the message is sent

### CODE REVIEW

File: src/main/java/net/cyberkitsune/prefixchat/ChatListener.java [hunk 1]
```diff
@@ -101,6 +103,14 @@ public void playerChat(AsyncPlayerChatEvent evt) {
...
 		}
 
+		// If message is longer than 200 characters (max is 256) reminder player they can append
+		if (evt.getMessage().length()>=200) {
+			if (!dashReminded.contains(evt.getPlayer().getDisplayName())) {
+				evt.getPlayer().sendMessage(LocalizedString.get("dashreminder", evt.getPlayer().getLocale()));
+				dashReminded.add(evt.getPlayer().getDisplayName());
+			}
+		}
+
```
Comment (cyberkitsune) [inline src/main/java/net/cyberkitsune/prefixchat/ChatListener.java:113]: This will fire if a user can't send a message, and before the message instead of after too. Move this block to right after line 129.

File: src/main/java/net/cyberkitsune/prefixchat/LocalizedString.java [hunk 2]
```diff
@@ -19,7 +21,7 @@ public static FileConfiguration getLocaleFile()
             KitsuneChat.getInstance().mcLog.severe("[KitsuneChat] Localization resource not found!!!");
             return null;
         }
-        InputStreamReader reader = new InputStreamReader(inputStream);
+        InputStreamReader reader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
```
Comment (cyberkitsune) [inline src/main/java/net/cyberkitsune/prefixchat/LocalizedString.java:24]: Unrelated fix. OK for now but good practice to make sure pull requests only address one issue

File: src/main/java/net/cyberkitsune/prefixchat/ChatListener.java [hunk 3]
```diff
@@ -18,8 +17,11 @@
 
 	public ChatListener() {
 		this.bufs = new HashMap<>();
+		dashReminded = new ArrayList<String>();
 	}
 
+	private List<String> dashReminded;
```
Comment (cyberkitsune) [inline src/main/java/net/cyberkitsune/prefixchat/ChatListener.java:23]: Users are added to list but never removed. Please find a place to remove users from list.

File: src/main/java/net/cyberkitsune/prefixchat/ChatListener.java (hunk 1 above)
Comment (sournub) [reply src/main/java/net/cyberkitsune/prefixchat/ChatListener.java:113]: It didn't matter if the block was at the top of everything, or below at the very bottom after the line where the line is sent, the reminder was still sent before the message was shown in the chat for the player.
2IrEMTR

File: src/main/java/net/cyberkitsune/prefixchat/ChatListener.java (hunk 3 above)
Comment (sournub) [reply src/main/java/net/cyberkitsune/prefixchat/ChatListener.java:24]: added playerquitevent to remove them from dashreminded list
